.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,7 @@
+- id: gofumpt-format
+  name: Go Code Formatter
+  description: "Automatically formats Go code using gofumpt. Requires Go and gofumpt to be installed."
+  entry: hooks/run-go-fumpt.sh
+  language: script
+  types: [go]
+  files: '\.go$'
```

Comment:
I would suggest using something like this

https://github.com/shutter-network/pre-commit-go-hooks/blob/main/.pre-commit-hooks.yaml

By using language and addition dependency